static/js/components/Loading.js

```diff
@@ -1,27 +1,47 @@
 // @flow
 import React from "react"
 
-import { anyProcessing, anyError, allLoaded } from "../util/rest"
-
-import type { RestState } from "../flow/restTypes"
-
-export default class Loading extends React.Component {
-  props: {
-    restStates: Array<RestState<*>>,
-    renderContents: Function
-  }
+type LoadingProps = {
+  loaded: boolean,
+  errored: boolean
+}
 
-  render() {
-    const { restStates, renderContents } = this.props
+const withLoading = (
+  LoadedComponent: Class<React.Component<*, *, *>>,
+  shouldUpdateRequirements: (Object, Object) => boolean,
+  updateRequirements: Object => void
+) => {
+  return class extends React.Component {
+    props: LoadingProps
 
-    if (anyProcessing(restStates) || !allLoaded(restStates)) {
-      return <div>Loading</div>
+    componentWillMount() {
+      updateRequirements(this.props)
     }
 
-    if (anyError(restStates)) {
-      return <div>Error loading page</div>
+    componentDidUpdate(prevProps: Object) {
+      if (shouldUpdateRequirements(prevProps, this.props)) {
+        updateRequirements(this.props)
+      }
```

Comment:
I know this was the convention before, I think `updateRequirements` and `shouldUpdateRequirements` are kind of awkward names here. What about `loadData` and `shouldLoadData`? To me that makes more sense in this context